APIContract/notifications-admin

diegolopezl opened the pull request:
Get notified when a post is liked, commented or has reached 100 likes also when you receive a friend request or a request is accepted and when you reach 25 friends . Admin can view, edit or delete a user's stats, posts, friends and groups.

diegolopezl committed:
APIContract/notifications-admin

Get notified when a post is liked, commented or has reached 100 likes also when you receive a friend request or a request is accepted and when you reach 25 friends . Admin can view, edit or delete a user's stats, posts, friends and groups.

kajahno commented on api-contract/contract.yaml:
```diff
@@ -512,7 +517,88 @@ paths:
+      operationId: seeStats
+      requestBody:
+        description: View admin object
+        content:
+          application/json:
+            schema:
+              type: array
+              $ref: '#/components/schemas/Admin'
+        required: true
```
I believe this is not going to work. Check it in the OpenAPI preview, I bet it doesn't render as it should.

kajahno commented on api-contract/contract.yaml:
```diff
@@ -512,7 +517,88 @@ paths:
+        description: Edit admin object
+        content:
+          application/json:
+            schema:
+              type: array
+              $ref: '#/components/schemas/Admin'
+        required: true
```
Same problem as above.

kajahno commented on api-contract/contract.yaml:
```diff
@@ -512,7 +517,88 @@ paths:
+      requestBody:
+        description: Delete admin object
+        content:
+          application/json:
+            schema:
+              type: array
+              $ref: '#/components/schemas/Admin'
+        required: true
```
ditto

kajahno commented on api-contract/contract.yaml:
```diff
@@ -654,6 +740,50 @@ components:
           type: string
         updatedAt:
           type: string
+    Notification:
+      type: object
+      properties:
+        liked:
+          type: string
+        reached100Likes:
+          type: string
+          items:
+            $ref: '#/components/schemas/Post'
+        reached25Friends:
+          type: string
+          items:
+            $ref: '#/components/schemas/UserProfile'
+        friendRequest:
+          type: string
+        requestAccepted:
+          type: string
+        commented:
+          type: string
+        userId:
+          type: string
+        notifId:
+          type: string
+        postID:
+          type: string
```
We need to think about this one again. Basically the point of the notification is to have:
* an id, notifId iD is not a good name, we should use something more obvious, like notificationId (also to be consistent with the other schemas)
* we need to have IDs of the schemas that we provide notifications for, e.g. user / post / comment and so on. I would implement it by just adding the properties: userId, postId, commentId
* we need to have a message, literally a property just named: 'message'
* we need to have a boolean to know if the notification was read or not, another property named 'read'

I think that's it pretty much

diegolopezl committed:
APIContract/notification-admin/adjustments

Added new properties to Notification schema and changed admin to show as an array

ssanabia2 committed:
APIContract/post-adjustments

New endpoint: /post/all
